src/main.rs

```diff
@@ -16,6 +16,13 @@
 #![allow(clippy::struct_excessive_bools)]
 #![allow(clippy::too_many_lines)]
 
+#[cfg(not(any(feature = "native-roots", feature = "webpki-roots")))]
+compile_error!("either feature \"native-roots\" or feature \"webpki-roots\" must be enabled");
+#[cfg(all(feature = "native-roots", feature = "webpki-roots"))]
+compile_error!(
+    "feature \"native-roots\" and feature \"webpki-roots\" cannot be enabled at the same time"
+);
+
```

Comment:
Good catch! How do you feel about one unified explanation like so:
```rust
#[cfg(any(
    all(feature = "native-roots", feature = "webpki-roots"),
    not(any(feature = "native-roots", feature = "webpki-roots")),
))]
compile_error!(
    "exactly one of feature \"native-roots\" and feature \"webpki-roots\" must be enabled"
);
```

I think both versions are fine though

Reply:
That looks nicer, yes!